quill-core/src/main/scala/io/getquill/norm/BetaReduction.scala

```diff
@@ -8,7 +8,12 @@ case class BetaReduction(map: collection.Map[Ident, Ast])
   override def apply(ast: Ast) =
     ast match {
       case Property(Tuple(values), name) =>
-        apply(values(name.drop(1).toInt - 1))
+        val aliases = values.distinct
+        aliases match{
```

Comment:
> Maybe we should have a new AST class for the new type of join?

I thought about that but it may complicate things more than necessary. As it stands a secondary join is represented as a self join in the AST, which is easily encoded in the DSL (by reusing existing `Join` class). 

Since `FreshIdent` is guaranteed to produce unique aliases we know in `BetaReduction.scala` if size of `values` is *greater* than 1 but the number of distinct `values` equals 1, we're working with a self join.

As for "Why do you need this transformation?", because it blows up without it ;-) See for yourself, grab the commits and test against local branch with just `apply(values(name.drop(1).toInt - 1))`, you'll get a `NumberFormatException`